main.py

```diff
@@ -168,10 +168,13 @@ def generate_bepress_text(baseurl, shortname):
     return simple_oai(class_name, shortname, longname, baseurl, prop_list, tz_gran)
 
 
-def generate_oai(baseurl, shortname):
-    prop_list = get_oai_properties(baseurl, shortname)
+def generate_oai(baseurl, shortname,start_date,end_date):
+    prop_list = get_oai_properties(baseurl, shortname,start_date,end_date)
     ex_call = baseurl + '?verb=ListRecords&metadataPrefix=oai_dc'
-    class_name = shortname.capitalize()
+    try:
+        class_name = shortname.capitalize()
+    except:
```

Comment:
catch a specific exception here - bare excepts are unclear as to what you're expecting to happen